### PR TITLE
remove robot accounts when to delete a project

### PR DESCRIPTION
**Pull request by wy65701436**
The robots associate with the project should be removed after the project is deleted.

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/pkg/robot/mock.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package robot
+
+//go:generate mockery --dir ./dao --name RobotAccountDao --output . --outpkg robot --filename mock_robot_dao_test.go --structname mockRobotDao
```
Comment (heww) [inline src/pkg/robot/mock.go:3]: Move it to `testing` pkg.

File: src/pkg/robot/dao/robot.go [hunk 2]
```diff
@@ -103,3 +108,17 @@ func (r *robotAccountDao) DeleteRobotAccount(id int64) error {
...
 	return err
 }
+
+// DeleteByProjectID ...
+func (r *robotAccountDao) DeleteByProjectID(ctx context.Context, projectID int64) error {
+	qs, err := libOrm.QuerySetter(ctx, &model.Robot{}, &q.Query{
+		Keywords: map[string]interface{}{
+			"ProjectID": projectID,
+		},
+	})
```
Comment (heww) [inline src/pkg/robot/dao/robot.go:118]:
```go
qs, err := libOrm.QuerySetter(ctx, &model.Robot{}, q.New(q.Keywords{"ProjectID": projectID}))
```

File: src/server/v2.0/handler/project.go [hunk 3]
```diff
@@ -190,6 +193,11 @@ func (a *projectAPI) DeleteProject(ctx context.Context, params operation.DeleteP
 		return a.SendError(ctx, errors.PreconditionFailedError(errors.New(result.Message)))
 	}
 
+	// remove the robot associated with the project
+	if err := a.robotMgr.DeleteByProjectID(ctx, params.ProjectID); err != nil {
+		return a.SendError(ctx, err)
+	}
+
```
Comment (heww) [inline src/server/v2.0/handler/project.go:200]: Please delete the robots after the project was deleted.

File: src/pkg/robot/mock.go (hunk 1 above)
Comment (wy65701436) [reply src/pkg/robot/mock.go]: updated.

**Pull request merged**